src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Join.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.PowerFx.Core.App.ErrorContainers;
+using Microsoft.PowerFx.Core.Errors;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Types.Enums;
+using Microsoft.PowerFx.Syntax;
+
+namespace Microsoft.PowerFx.Core.Texl.Builtins
+{
+    internal class JoinFunction : FilterFunctionBase
+    {
+        public override bool IsSelfContained => true;
+
+        public override int ScopeArgs => 2;
+
+        public JoinFunction()
+            : base("Join", TexlStrings.AboutJoin, FunctionCategories.Table, DType.EmptyTable, 0x8, 3, 4, DType.EmptyTable, DType.EmptyTable)
+        {
+            ScopeInfo = new FunctionJoinScopeInfo(this);
+        }
+
+        public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
+        {
+            yield return new[] { TexlStrings.JoinArg1, TexlStrings.JoinArg2 };
```

Comment:
If the minimum arity is 3 (from line 23), we should not suggest a signature with 2 elements only